homeassistant/components/ssdp/__init__.py

```diff
@@ -61,55 +64,67 @@
     "location": ATTR_SSDP_LOCATION,
 }
 
+SsdpChange = Enum("SsdpChange", "ALIVE BYEBYE UPDATE")
+SsdpCallback = Callable[[Mapping[str, Any], SsdpChange], Awaitable]
+
+
+SSDP_SOURCE_SSDP_CHANGE_MAPPING: Mapping[SsdpSource, SsdpChange] = {
+    SsdpSource.SEARCH: SsdpChange.ALIVE,
+    SsdpSource.ADVERTISEMENT: SsdpChange.ALIVE,
```

Comment:
Can we eliminate the non-exact case? Is there ever a case where we don't know if its an `Alive/ByeBye/Update`?

Reply:
Can you elaborate on this?

Reply:
Can we eliminate the `SsdpSource.ADVERTISEMENT` case since we always treat them as alive anyways?